### PR TITLE
Bootstrapper updates

### LINKED ISSUE #6358: Update-AzureRmProfile - Error: UseBasicParsing
### Description

While trying steps following: https://docs.microsoft.com/en-us/azure/azure-stack/user/azure-stack-version-profiles-powershell.

Having multiple issues:
1. (get-azurermprofile).module claims that not all modules are installed
2. Install-AzureAzureRmProfile does not seem to work/respond at all/do what it is supposed to do
3. Update-AzureRmProfile throws errors on needing Internet Explorer. 

Check Reproduction

### Script/Steps for Reproduction
```powershell
PS C:\WINDOWS\system32> install-module azurerm.bootstrapper

Untrusted repository
You are installing the modules from an untrusted repository. If you trust this repository, change its
InstallationPolicy value by running the Set-PSRepository cmdlet. Are you sure you want to install the modules from
'PSGallery'?
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "N"): y
PS C:\WINDOWS\system32> Install-AzureRMProfile -Profile 2017-03-09-profile

Untrusted repository
You are installing the modules from an untrusted repository. If you trust this repository, change its
InstallationPolicy value by running the Set-PSRepository cmdlet. Are you sure you want to install the modules from
'PSGallery'?
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "N"): y
PS C:\WINDOWS\system32> get-azurermprofile


ProfileName : 2017-03-09-profile
Modules     : {AzureRM : 1.2.11
              , AzureRM.Compute : 1.2.3.4
              , AzureRM.Storage : 1.0.5.4
              }

WARNING: Some modules from profile(s) latest were not installed. Use Install-AzureRmProfile to install missing modules.


PS C:\WINDOWS\system32> (get-azurermprofile).modules
WARNING: Some modules from profile(s) latest were not installed. Use Install-AzureRmProfile to install missing modules.
PS C:\WINDOWS\system32> get-package azurerm.storage

Name                           Version          Source                           ProviderName
----                           -------          ------                           ------------
AzureRM.Storage                1.0.5.4          https://www.powershellgallery... PowerShellGet


PS C:\WINDOWS\system32> get-package azurerm

Name                           Version          Source                           ProviderName
----                           -------          ------                           ------------
AzureRM                        1.2.11           https://www.powershellgallery... PowerShellGet


PS C:\WINDOWS\system32> Install-AzureRMProfile -Profile 2017-03-09-profile
PS C:\WINDOWS\system32> (get-azurermprofile).modules
WARNING: Some modules from profile(s) latest were not installed. Use Install-AzureRmProfile to install missing modules.
PS C:\WINDOWS\system32> Install-AzureRMProfile -Profile 2017-03-09-profile
PS C:\WINDOWS\system32> Update-AzureRmProfile -Profile 2017-03-09-profile
Invoke-CommandWithRetry : The response content cannot be parsed because the Internet Explorer engine is not available,
or Internet Explorer's first-launch configuration is not complete. Specify the UseBasicParsing parameter and try
again.
At C:\Program Files\WindowsPowerShell\Modules\AzureRM.BootStrapper\0.3.0\AzureRM.BootStrapper.psm1:104 char:18
+   $WebResponse = Invoke-CommandWithRetry -ScriptBlock $ScriptBlock
+                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotImplemented: (:) [Invoke-CommandWithRetry], NotSupportedException
    + FullyQualifiedErrorId : WebCmdletIEDomNotSupportedException,Invoke-CommandWithRetry

Cannot index into a null array.
At C:\Program Files\WindowsPowerShell\Modules\AzureRM.BootStrapper\0.3.0\AzureRM.BootStrapper.psm1:118 char:3
+   $OnlineProfileMapETag = $WebResponse.Headers["ETag"]
+   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray

ConvertFrom-Json : Cannot bind argument to parameter 'InputObject' because it is null.
At C:\Program Files\WindowsPowerShell\Modules\AzureRM.BootStrapper\0.3.0\AzureRM.BootStrapper.psm1:172 char:38
+   $OnlineProfileMap = $WebResponse | ConvertFrom-Json
+                                      ~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [ConvertFrom-Json], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.ConvertFrom
   JsonCommand

Loading Profile 2017-03-09-profile
```

### Module Version

```powershell
Get-Module -Name AzureRM -ListAvailable
    Directory: C:\Program Files\WindowsPowerShell\Modules
ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     1.2.11     AzureRM
```

### Environment Data

```powershell
$PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.17134.48
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.17134.48
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1


```

### Debug Output

```powershell
PS C:\WINDOWS\system32> $DebugPreference = "Continue"
PS C:\WINDOWS\system32> Update-AzureRmProfile -Profile 2017-03-09-profile
Invoke-CommandWithRetry : The response content cannot be parsed because the Internet Explorer engine is not available,
or Internet Explorer's first-launch configuration is not complete. Specify the UseBasicParsing parameter and try
again.
At C:\Program Files\WindowsPowerShell\Modules\AzureRM.BootStrapper\0.3.0\AzureRM.BootStrapper.psm1:104 char:18
+   $WebResponse = Invoke-CommandWithRetry -ScriptBlock $ScriptBlock
+                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotImplemented: (:) [Invoke-CommandWithRetry], NotSupportedException
    + FullyQualifiedErrorId : WebCmdletIEDomNotSupportedException,Invoke-CommandWithRetry

Cannot index into a null array.
At C:\Program Files\WindowsPowerShell\Modules\AzureRM.BootStrapper\0.3.0\AzureRM.BootStrapper.psm1:118 char:3
+   $OnlineProfileMapETag = $WebResponse.Headers["ETag"]
+   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray

ConvertFrom-Json : Cannot bind argument to parameter 'InputObject' because it is null.
At C:\Program Files\WindowsPowerShell\Modules\AzureRM.BootStrapper\0.3.0\AzureRM.BootStrapper.psm1:172 char:38
+   $OnlineProfileMap = $WebResponse | ConvertFrom-Json
+                                      ~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [ConvertFrom-Json], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.ConvertFrom
   JsonCommand

Loading Profile 2017-03-09-profile

```

### PR DESCRIPTION
**Pull request by viananth**
1. Decoupled Remove-previousversions and added Set-ProfilemapEndpoint to support disconnected scenarios
2. Get-AzureRmModule returns module object
3. Added -UseBasicParsing to avoid IE error
4. Test updates

Closes #6358, #4274, #4335

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: tools/AzureRM.BootStrapper/AzureRM.Bootstrapper.psm1 [hunk 1]
```diff
@@ -88,7 +88,7 @@ $script:LatestProfileMapPath = Get-LatestProfileMapPath
 function Get-AzureStorageBlob
 {
   $ScriptBlock = {
-    Invoke-WebRequest -uri $PSProfileMapEndpoint -ErrorVariable RestError
+    Invoke-WebRequest -uri $Script:PSProfileMapEndpoint -UseBasicParsing -ErrorVariable RestError
```
Comment (deathly809) [inline tools/AzureRM.BootStrapper/AzureRM.Bootstrapper.psm1]: What do you think of moving to using WebClient?  We find it to be much faster.

**Comment by markcowl** [comment]
@viananth This is out of scope for the current release.  Closing for now.

**Comment by deathly809** [comment]
@markcowl  This is for January release.  Krishna want's this to go in, what is the issue?

**Comment by markcowl** [comment]
@viananth @deathly809 There is no client side update planned until June. We don't want to spend cycles on this when it will be replaced by multi-api versions.  Unless there is a critical customer need,we shoudl defer.

**Pull request closed without merging**